code/datums/datum.dm

```diff
@@ -58,6 +58,15 @@
 
 	return QDEL_HINT_QUEUE
 
+/datum/proc/CalcMemory()
+	. = 24
+	var/list/builtins = list(NAMEOF(src, type) = TRUE, NAMEOF(src, vars) = TRUE, NAMEOF(src, tag) = TRUE, NAMEOF(src, parent_type) = TRUE)
+	for(var/I in vars)
+		if(builtins[I])

```

Comment:
It might be faster to do `vars-builtin`, allowing you to forgo the associated part. Look at whatever I did in the findReferences TESTING proc in ssgarbage for the var blacklist there, as that was what profiled fastest.